Git-commit: 5478e9e33b6058af56171ed272f05a61d2db5efc
From: Sang Yan <sangyan(a)huawei.com>
Date: Wed, 24 Mar 2021 15:29:17 +0800
Subject: [PATCH] arm64: amend config position of park and pmem
Patch-mainline: Queued in openEuler repo, version 5.10.0-4.19.0
References: openEuler-21.03
Git-repo: https://gitee.com/openeuler/kernel.git

hulk inclusion
category: feature
bugzilla: 48159
CVE: N/A

------------------------------

Move config of ARM64_CPU_PARK into menu "Kernel features"
and beside CRASH_DUMP.

Move config of ARM64_PMEM_RESERVE ARM64_PMEM_LEGACY_DEVICE
into menu "Kernel features".

Signed-off-by: Sang Yan <sangyan(a)huawei.com>
Reviewed-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 arch/arm64/Kconfig                     | 66 +++++++++++++-------------
 arch/arm64/configs/openeuler_defconfig |  6 +--
 2 files changed, 36 insertions(+), 36 deletions(-)

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 326f26d40f7f..72c7ce399d8b 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -345,42 +345,9 @@ config KASAN_SHADOW_OFFSET
 	default 0xeffffff900000000 if ARM64_VA_BITS_36 && KASAN_SW_TAGS
 	default 0xffffffffffffffff
 
-config ARM64_CPU_PARK
-	bool "Support CPU PARK on kexec"
-	depends on SMP
-	depends on KEXEC_CORE
-	help
-	 This enables support for CPU PARK feature in
-	 order to save time of cpu down to up.
-	 CPU park is a state through kexec, spin loop
-	 instead of cpu die before jumping to new kernel,
-	 jumping out from loop to new kernel entry in
-	 smp_init.
-
 config ARCH_HAS_CPU_RELAX
 	def_bool y
 
-config ARM64_PMEM_RESERVE
-	bool "Reserve memory for persistent storage"
-	default n
-	help
-	  Use memmap=nn[KMG]!ss[KMG](memmap=100K!0x1a0000000) reserve
-	  memory for persistent storage.
-
-	  Say y here to enable this feature.
-
-config ARM64_PMEM_LEGACY_DEVICE
-	bool "Create persistent storage"
-	depends on BLK_DEV
-	depends on LIBNVDIMM
-	select ARM64_PMEM_RESERVE
-	help
-	  Use reserved memory for persistent storage when the kernel
-	  restart or update. the data in PMEM will not be lost and
-	  can be loaded faster.
-
-	  Say y if unsure.
-
 source "arch/arm64/Kconfig.platforms"
 
 menu "Kernel Features"
@@ -1196,6 +1163,18 @@ config CRASH_DUMP
 
 	  For more details see Documentation/admin-guide/kdump/kdump.rst
 
+config ARM64_CPU_PARK
+	bool "Support CPU PARK on kexec"
+	depends on SMP
+	depends on KEXEC_CORE
+	help
+	 This enables support for CPU PARK feature in
+	 order to save time of cpu down to up.
+	 CPU park is a state through kexec, spin loop
+	 instead of cpu die before jumping to new kernel,
+	 jumping out from loop to new kernel entry in
+	 smp_init.
+
 config XEN_DOM0
 	def_bool y
 	depends on XEN
@@ -1257,6 +1236,27 @@ config RODATA_FULL_DEFAULT_ENABLED
 	  This requires the linear region to be mapped down to pages,
 	  which may adversely affect performance in some cases.
 
+config ARM64_PMEM_RESERVE
+	bool "Reserve memory for persistent storage"
+	default n
+	help
+	  Use memmap=nn[KMG]!ss[KMG](memmap=100K!0x1a0000000) reserve
+	  memory for persistent storage.
+
+	  Say y here to enable this feature.
+
+config ARM64_PMEM_LEGACY_DEVICE
+	bool "Create persistent storage"
+	depends on BLK_DEV
+	depends on LIBNVDIMM
+	select ARM64_PMEM_RESERVE
+	help
+	  Use reserved memory for persistent storage when the kernel
+	  restart or update. the data in PMEM will not be lost and
+	  can be loaded faster.
+
+	  Say y if unsure.
+
 config ARM64_SW_TTBR0_PAN
 	bool "Emulate Privileged Access Never using TTBR0_EL1 switching"
 	help
diff --git a/arch/arm64/configs/openeuler_defconfig b/arch/arm64/configs/openeuler_defconfig
index 21c5d1125f8d..1aedaf3c61aa 100644
--- a/arch/arm64/configs/openeuler_defconfig
+++ b/arch/arm64/configs/openeuler_defconfig
@@ -275,9 +275,6 @@ CONFIG_FIX_EARLYCON_MEM=y
 CONFIG_PGTABLE_LEVELS=3
 CONFIG_ARCH_SUPPORTS_UPROBES=y
 CONFIG_ARCH_PROC_KCORE_TEXT=y
-CONFIG_ARM64_CPU_PARK=y
-CONFIG_ARM64_PMEM_RESERVE=y
-CONFIG_ARM64_PMEM_LEGACY_DEVICE=y
 
 #
 # Platform selection
@@ -407,10 +404,13 @@ CONFIG_PARAVIRT_TIME_ACCOUNTING=y
 CONFIG_KEXEC=y
 # CONFIG_KEXEC_FILE is not set
 CONFIG_CRASH_DUMP=y
+CONFIG_ARM64_CPU_PARK=y
 # CONFIG_XEN is not set
 CONFIG_FORCE_MAX_ZONEORDER=14
 CONFIG_UNMAP_KERNEL_AT_EL0=y
 CONFIG_RODATA_FULL_DEFAULT_ENABLED=y
+CONFIG_ARM64_PMEM_RESERVE=y
+CONFIG_ARM64_PMEM_LEGACY_DEVICE=y
 # CONFIG_ARM64_SW_TTBR0_PAN is not set
 CONFIG_ARM64_TAGGED_ADDR_ABI=y
 CONFIG_ARM64_ILP32=y
-- 
2.31.1

